src/TS_HW3/Accounting.ts

```diff
@@ -0,0 +1,59 @@
+/*
+    Так само у нас є сутність Бухгалтерія, яка є департаментом і має властивість баланс, а також методи
+    для взяття на баланс співробітника або департаменту, зняття з балансу і виплати зарплати для всього персоналу.
+    Попередньо найняті співробітники отримують зарплату за допомогою зовнішніх оплат, Співробітники (тільки активні) - за допомогою внутрішніх.
+*/
+class Accounting extends Department {
+  public Balance: number;
+
+  constructor(company: Company, domain: string, debit: number, credit: number) {
+    super(company, DepartmentsNames.Accounting, domain, debit, credit);
+    this.Balance = debit - credit;
+  }
+
+  public TakeOnBalance(item: Employee | PreEmployee | Department): void {
+    switch (true) {
+      case item instanceof Employee || item instanceof PreEmployee: {
+        this.Balance += item.Salary;
+        break;
+      }
+      default: {
+        this.Balance += item.GetBalance();
+      }
+    }
+  }
+
+  public TakeOffBalance(item: Employee | PreEmployee | Department): void {
+    switch (true) {
```

Comment:
It may not be the best approach to use a 'switch' statement in this case.

Reply:
ok, the same as above